File: Medium/19. Remove Nth Node From End of List/Solution.java
//   Definition for singly-linked list.
//   public class ListNode {
//   int val;
//   ListNode next;
//   ListNode() {}
//   ListNode(int val) { this.val = val; }
//   ListNode(int val, ListNode next) { this.val = val; this.next = next; }
//   }
 
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode traverse = head;
        while(n!=0){
            traverse = traverse.next;
            n--;
        }

        if(traverse == null) return head.next;

        traverse = traverse.next;
        ListNode prev = head;
        while(traverse != null){
            prev = prev.next;
            traverse = traverse.next;
        }
        
        prev.next = prev.next.next;
        return head;
    }
}

File: Medium/19. Remove Nth Node From End of List/Solution.cpp
#include <stdlib.h>

// Definition for singly-linked list.
// struct ListNode {
//     int val;
//     ListNode *next;
//     ListNode() : val(0), next(nullptr) {}
//     ListNode(int x) : val(x), next(nullptr) {}
//     ListNode(int x, ListNode *next) : val(x), next(next) {}
// };

class Solution
{
public:
    ListNode *removeNthFromEnd(ListNode *head, int n)
    {
        ListNode *traverse = head;
        while (n != 0)
        {
            traverse = traverse->next;
            n--;
        }

        if (traverse == NULL)
            return head->next;

        traverse = traverse->next;
        ListNode *prev = head;
        while (traverse != NULL)
        {
            prev = prev->next;
            traverse = traverse->next;
        }

        prev->next = prev->next->next;
        return head;
    }
};